config.go

```diff
@@ -37,48 +37,62 @@ import (
 )
 
 const (
-	defaultConfigFilename        = "dcrd.conf"
-	defaultDataDirname           = "data"
-	defaultLogLevel              = "info"
-	defaultLogDirname            = "logs"
-	defaultLogFilename           = "dcrd.log"
-	defaultMaxSameIP             = 5
-	defaultMaxPeers              = 125
-	defaultBanDuration           = time.Hour * 24
-	defaultBanThreshold          = 100
-	defaultMaxRPCClients         = 10
-	defaultMaxRPCWebsockets      = 25
-	defaultMaxRPCConcurrentReqs  = 20
-	defaultDbType                = "ffldb"
+	// Defaults for general application behavior options.
+	defaultConfigFilename  = "dcrd.conf"
+	defaultDataDirname     = "data"
+	defaultLogDirname      = "logs"
+	defaultLogFilename     = "dcrd.log"
+	defaultDbType          = "ffldb"
+	defaultLogLevel        = "info"
+	defaultSigCacheMaxSize = 100000
+
+	// Defaults for RPC server options and policy.
+	defaultTLSCurve             = "P-521"
+	defaultMaxRPCClients        = 10
+	defaultMaxRPCWebsockets     = 25
+	defaultMaxRPCConcurrentReqs = 20
+
+	// Defaults for P2P network options.
+	defaultMaxSameIP       = 5
+	defaultMaxPeers        = 125
+	defaultDialTimeout     = time.Second * 30
+	defaultPeerIdleTimeout = time.Second * 120
+
+	// Defaults for banning options.
+	defaultBanDuration  = time.Hour * 24
+	defaultBanThreshold = 100
+
+	// Defaults for relay and mempool policy options.
 	defaultFreeTxRelayLimit      = 15.0
-	defaultBlockMinSize          = 0
-	defaultBlockMaxSize          = 375000
-	blockMaxSizeMin              = 1000
-	defaultAddrIndex             = false
-	defaultGenerate              = false
-	defaultNoMiningStateSync     = false
-	defaultAllowUnsyncedMining   = false
-	defaultAllowOldVotes         = false
 	defaultMaxOrphanTransactions = 100
-	defaultMaxOrphanTxSize       = mempool.MaxStandardTxSize
-	defaultSigCacheMaxSize       = 100000
-	defaultTxIndex               = false
-	defaultNoExistsAddrIndex     = false
-	defaultNoCFilters            = false
-	defaultTLSCurve              = "P-521"
-	defaultDialTimeout           = time.Second * 30
-	defaultPeerIdleTimeout       = time.Second * 120
+	defaultAllowOldVotes         = false
+
+	// Defaults for mining options and policy.
+	defaultGenerate            = false
+	defaultBlockMinSize        = 0
+	defaultBlockMaxSize        = 375000
+	blockMaxSizeMin            = 1000
+	defaultNoMiningStateSync   = false
+	defaultAllowUnsyncedMining = false
+
+	// Defaults for indexing options.
+	defaultTxIndex           = false
+	defaultAddrIndex         = false
+	defaultNoExistsAddrIndex = false
+	defaultNoCFilters        = false
```

Comment:
I realize this isn't a change, but I am confused about the need for default, false bools. Not all bools are covered, and the default is false anyway.

Reply:
They're not really required, but the original intention was that all config options had a default so it was easy to change them.  I didn't want to change anything other than categorization in this PR, so I left them.